docs/sources/reference/glossary.md

```diff
@@ -0,0 +1,194 @@
+normally execute manually in order to build a Docker image. Docker can
+build images automatically by reading the instructions from a Dockerfile.
+
+## filesystem
+
+A file system is the method an operating system uses to name files
+and assign them locations for efficient storage and retrieval.
+
+Examples :
+
+Linux : ext4, aufs, btrfs, zfs
+Windows : NTFS
+OS X : HFS+
+
+## image
+
+docker images are the basis of [containers](#container). An Image is an
+ordered collection of root filesystem changes and the corresponding
+execution parameters for use within a container runtime. An image typically
+contains a union of layered filesystems stacked on top of each other. An image
+does not have state and it never changes.
+
+## libcontainer
+
+libcontainer provides a native Go implementation for creating containers with
+namespaces, cgroups, capabilities, and filesystem access controls. It allows
+you to manage the lifecycle of the container performing additional operations
+after the container is created.
+
+## link
+
+links provide an interface to connect docker containers running on the same host
+to each other without exposing the hosts' network ports. When you set up a link,
+you create a conduit between a source container and a recipient container.
+The recipient can then access select data about the source. To create a link,
+you can use the `--link` flag.
+
+## machine
+
+machine is a docker tool which makes it really easy to create docker hosts on 
+your computer, on cloud providers and inside your own data center. It creates
+servers, installs Docker on them, then configures the docker client to talk to them.
+
+*Also known as : docker-machine*
+
+## overlay
+
+OverlayFS is a [filesystem](#filesystem) service for Linux which implements a
+[union mount](http://en.wikipedia.org/wiki/Union_mount) for other file systems.
+It is supported by the docker daemon as a storage driver.
+
+## registry
+
+A Registry is a hosted service containing [repositories](#repository) of [images](#image)
+which responds to the Registry API.
+
+The default registry can be accessed using a browser at [Docker Hub](#Docker-Hub)
+or using the `docker search` command.
+
+## repository
+
+A repository is a set of docker images. A repository can be shared by pushing it
+to a [registry](#registry) server. The different images in the repository can be
+labeled using [tags](#tag). 
+
+Here is an example of the shared [nginx repository](https://registry.hub.docker.com/_/nginx/)
+and its [tags](https://registry.hub.docker.com/_/nginx/tags/manage/)
+
+## swarm
+
+swarm is a native clustering tool for docker. Swarm pools together several docker hosts
+and exposes them as a single virtual docker host. It serves the standard Docker API,
+so any tool that already works with Docker can now transparently scale up to multiple hosts.
+
+*Also known as : docker-swarm*
+
+## tag
+
+A tag is a label applied to a docker image in a [repository](#repository).
```

Comment:
Might need to be careful using "label", because Docker now also has "labels" as a concept (you can now add labels to images and containers)